qiskit/algorithms/minimum_eigensolvers/vqe.py

```diff
@@ -0,0 +1,333 @@
+                :class:`.Minimizer` protocol.
+            gradient: An optional estimator gradient to be used with the optimizer.
+            initial_point: An optional initial point (i.e. initial parameter values) for the optimizer.
+                The length of the initial point must match the number of :attr:`ansatz` parameters.
+                If ``None``, a random point will be generated within certain parameter bounds.
+                ``VQE`` will look to the ansatz for these bounds. If the ansatz does not specify
+                bounds, bounds of :math:`-2\pi`, :math:`2\pi` will be used.
+            callback: A callback that can access the intermediate data at each optimization step.
+                These data are: the evaluation count, the optimizer parameters for the ansatz, the
+                estimated value, and the metadata dictionary.
+        """
+        super().__init__()
+
+        self.estimator = estimator
+        self.ansatz = ansatz
+        self.optimizer = optimizer
+        self.gradient = gradient
+        # this has to go via getters and setters due to the VariationalAlgorithm interface
+        self.initial_point = initial_point
+        self.callback = callback
+
+    @property
+    def initial_point(self) -> Sequence[float] | None:
+        return self._initial_point
+
+    @initial_point.setter
+    def initial_point(self, value: Sequence[float] | None) -> None:
+        self._initial_point = value
+
+    def compute_minimum_eigenvalue(
+        self,
+        operator: BaseOperator | PauliSumOp,
+        aux_operators: ListOrDict[BaseOperator | PauliSumOp] | None = None,
+    ) -> VQEResult:
+        self._check_operator_ansatz(operator)
+
+        initial_point = validate_initial_point(self.initial_point, self.ansatz)
+
+        bounds = validate_bounds(self.ansatz)
+
+        start_time = time()
+
+        evaluate_energy = self._get_evaluate_energy(self.ansatz, operator)
+
+        if self.gradient is not None:
+            evaluate_gradient = self._get_evaluate_gradient(self.ansatz, operator)
+        else:
+            evaluate_gradient = None
+
+        # perform optimization
+        if callable(self.optimizer):
+            optimizer_result = self.optimizer(
+                fun=evaluate_energy, x0=initial_point, jac=evaluate_gradient, bounds=bounds
+            )
+        else:
+            optimizer_result = self.optimizer.minimize(
+                fun=evaluate_energy, x0=initial_point, jac=evaluate_gradient, bounds=bounds
+            )
+
+        eval_time = time() - start_time
+
+        logger.info(
+            "Optimization complete in %s seconds.\nFound optimal point %s",
+            eval_time,
+            optimizer_result.x,
+        )
+
+        if aux_operators is not None:
+            bound_ansatz = self.ansatz.bind_parameters(optimizer_result.x)
+            aux_values = estimate_observables(self.estimator, bound_ansatz, aux_operators)
```

Comment:
It would be better if we could just pass the same ansatz we have been using and parameters for it to the estimate observables. If we bind the anstaz here it will be seen as a new circuit that needs to be transpiled etc rather than using a circuit it knows about and has been using for the main operator here and just being bound to the final parameters.

Reply:
Implemented :)

Reply:
Thanks!